osquery/tables/system/linux/md_tables.cpp

```diff
@@ -396,7 +398,8 @@ static inline bool handleMDStatuses(
     const std::string& searchTerm,
     std::function<void(const std::string& line)> handleFunc) {
   std::size_t pos = lines[n + 1].find(searchTerm);
-  if (pos != std::string::npos) {
+  if (pos != std::string::npos &&
+      lines[n + 1].length() > (pos + searchTerm.length())) {
```

Comment:
There's a while loop condition that checks n against vector boundary, and that being the only place the function is called, so it really shouldn't happen.  But I do agree this is bad design, as `handleMDStatuses` does nothing to ensure the caller gives it a `n` with a value that will not go out of bounds if incremented.  I'll restructure the code in a less error-prone way.